### PR TITLE
[CodeStyle][CINN] fix Ruff lint errors (flake8-comprehensions rules)

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Description

* https://github.com/PaddlePaddle/Paddle/issues/54953
fix Ruff flake8-comprehensions errors `C408`,`C416`,`C417`

C408:
fix 30
dict -> {} 20
list - > [] 10

C416:
fix 1
[line for line in sys.stdin] -> list(sys.stdin)

C417:
fix 2

### CODE REVIEW

**Comment by SigureMo** [comment]
image

File: pyproject.toml [hunk 1]
```diff
@@ -110,8 +110,8 @@ ignore = [
     "F811",
     "F821",
     "F901",
-    "C408",
-    "C417",
```
Comment (SigureMo) [inline pyproject.toml:111]: 直接删掉，不要注释

Comment (enkilee) [reply pyproject.toml:111]: 收到

File: pyproject.toml [hunk 2]
```diff
@@ -109,14 +109,11 @@ ignore = [
...
     "PLR0402",
     "PLC0414",
     "PLE1205",
 ]
 "tools/cinn/**" = [
     "F401",
-    "C416",
     "PLR0402",
-]
+]
```
Comment (SigureMo) [inline pyproject.toml]: pre-commit fix 下，这个文件代码风格明显不过

Comment (enkilee) [reply pyproject.toml]: OK

**Review by SigureMo — DISMISSED** [review]
LGTM

需要 rerun 下 PR-CI-Coverage

**Comment by SigureMo** [comment]
麻烦解决下冲突

**Comment by enkilee** [comment]
已解决冲突

**Review by SigureMo — APPROVED** [review]
LGTM

**Pull request merged**